src/Simulation/Native/src/simulator/wavefunction.hpp

```diff
@@ -90,6 +90,53 @@ class Wavefunction
 
     void flush() const
     {
+        //logic to reorder
+        Fusion fg = fused_.getFusedGates();
+        auto items = fg.getItems();
+        auto sets = fg.getSet();
+        auto ctrlSets = fg.getCtrlSet();
+        //getting all qubits to move to lower end of the wfn
+        if (items.size() > 0) {
+            std::vector<unsigned> indices;
+            std::set<unsigned> indicesSet;
```

Comment:
std::unordered_set might be a better choice here

Reply:
Just curious to understand the reason behind this recommendation?

Reply:
same as for map versus unordered_map -- it's likely to be more efficient